internal/fs/type_cache_test.go

```diff
@@ -0,0 +1,390 @@
+// helpers
+// //////////////////////////////////////////////////////////////////////
+func (t *typeCacheTestCommon) testNoInsertion() {
+	const name1 = "foo"
+	const contents = "taco"
+	var fi fs.FileInfo
+	var err error
+
+	// Create a file object in GCS.
+	fileObject, err := storageutil.CreateObject(
+		ctx,
+		bucket,
+		name1,
+		[]byte(contents))
+
+	ExpectEq(nil, err)
+	ExpectNe(nil, fileObject)
+
+	// Stat-call with file object. It should
+	// pass stat call, bypassing type-cache, as a file.
+	fi, err = os.Stat(path.Join(mntDir, name1))
+
+	ExpectEq(nil, err)
+	AssertNe(nil, fi)
+	ExpectFalse(fi.IsDir())
+
+	// Create a directory object in GCS with same name as the file object.
+	dirObject, err := storageutil.CreateObject(
+		ctx,
+		bucket,
+		name1+"/",
+		[]byte(contents))
+
+	ExpectEq(nil, err)
+	ExpectNe(nil, dirObject)
+
+	// Stat-call with directory object. It should
+	// pass stat call, bypassing type-cache, as a directory.
+	// It works because no entries are inserted in type-cache
+	// in this case.
+	fi, err = os.Stat(path.Join(mntDir, name1) + "/")
+
+	ExpectEq(nil, err)
+	AssertNe(nil, fi)
+	ExpectTrue(fi.IsDir())
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Tests for TypeCacheTestWithMaxEntries1
+// //////////////////////////////////////////////////////////////////////
+func (t *TypeCacheTestWithMaxSize1MB) TestSizeBasedEviction() {
+	const name1 = "foo"
+	const contents = "taco"
+	var fi fs.FileInfo
+	var err error
+
+	// Initially, without any existing object, type-cache
+	// should not contain any entry and os.Stat should fail.
+	_, err = os.Stat(path.Join(mntDir, name1))
+
+	ExpectNe(nil, err)
+
+	// Create a file object in GCS.
+	fileObject, err := storageutil.CreateObject(
+		ctx,
+		bucket,
+		name1,
+		[]byte(contents))
+
+	ExpectEq(nil, err)
+	ExpectNe(nil, fileObject)
+
+	// Stat-call with first file object. It should
+	// pass stat call, through type-cache as a file.
+	fi, err = os.Stat(path.Join(mntDir, name1))
+
+	ExpectEq(nil, err)
+	AssertNe(nil, fi)
+	ExpectFalse(fi.IsDir())
+
+	// Create a directory object in GCS with same name as the first file object.
+	dirObject, err := storageutil.CreateObject(
+		ctx,
+		bucket,
+		name1+"/",
+		[]byte(contents))
+
+	ExpectEq(nil, err)
+	ExpectNe(nil, dirObject)
+
+	// Stat-call with the directory object. It should
+	// fail as there is currently an entry for the first
+	// file object, which has the same name.
+	_, err = os.Stat(path.Join(mntDir, name1) + "/")
+
+	ExpectNe(nil, err)
+	ExpectThat(err, oglematchers.Error(oglematchers.HasSubstr("not a directory")))
+
+	// type-cache-max-size-mb = 1MiB = 1048576,
+	// and each entry-size = 92 (80 fixed + 12 key-length) Bytes.
+	// let's add another 11397(1048576/92) entries to evict
+	// the first file object ("foo") entry.
+	for i := 0; i < 11397; i++ {
+		name := fmt.Sprintf("object%06d", i)
+
+		// Create a new file object in GCS.
+		fileObject, err = storageutil.CreateObject(
```

Comment:
As discussed offline, it takes around 15 seconds. Can we try with `storageutil.CreateObjects`?

Reply:
I tried doing this. But realized that `storageutil.CreateObjects` doesn't return the created objects, without which I can't really use it here. I checked, CreateObjects is used in another 8-10 test files, so if I change the signature of storageutil.CreateObjects, then I'll have to make changes in all those files. Should I make the change in all these 9-10 files ?